### ISSUE TITLE
Massive Virtual Memory over-allocation 

### ISSUE THREAD

**Issue by domdevitto** [ghostery/ghostery-extension#475]
### Description
When adding/using release (8.4.2) extension from the Firefox extension site, to Firefox 70.0.1 (64bit) for MacOS, on MacOS 10.11.6 - the Virtual memory used by this extension (within the firefox plugin process) goes from 6gb to 12gb or even as high as 36Gb.

Before uninstall/re-install it was jumping to 36Gb of memory - even on a system disconnected from the internet on startup. 

It seems like this extension is allocating a massive data structure, and then never using it - however this distresses systems at startup (given the VM still needs to be allocated & cleared).

Upon use resident memory is a few hundred megs, but virtual remains 2x-4x the expected size, and the browser/system is consequently very laggy.

[Description of the bug or feature]

### Expected Behavior 

Disabling returns virtual memory use for the plugin process back to 6Gb, from 36/12 Gb.

[What you expected to happen]

### Actual Behavior 

Massive allocation of virtual memory, and browser/system slowdown which allocation occurs & while browsing.

[What actually happened]

### Steps to Reproduce

1. [First Step]
2. [Second Step]
3. [and so on...]

### Versions

+ Browser: Firefox 70.0.1 (64 bit)  (standard release)
+ OS:  MacOS 10.11.6
+ Node: n/a
+ NPM: n/a

**Comment by christophertino** [comment]
Is this from the FirefoxCP WebExtensions process? I'm showing 13GB of virtual memory on my system as well. That seems excessive.

**Comment by philipp-classen** [comment]
I'm closing it because it is old and high virtual memory should not a problem in itself. Only once the process starts using it, the operating system has to allocate physical memory.

If there are performance problems with Firefox, it is most likely caused by something else. Feel free to comment or open a new issue if you know how to reproduce.

---

Note: On Chrome, we recently closed a memory leak where the physical (not virtual) memory usage increased over time: https://github.com/ghostery/ghostery-extension/issues/481
